js/src/base-component.js

```diff
@@ -34,6 +39,18 @@ class BaseComponent {
     this._element = null
   }
 
+  _executeCallback(callback, element, isAnimated = false) {
```

Comment:
Personally I would call it `_queueCallback` or `_afterTransition` or something like that because you're not just executing a callback here. And maybe make isAnimated true by default.

Reply:
A small thing, but it is better to include `transitionEnd` in the name of the method. The name is not explanatory IMO :slightly_smiling_face:
May be `_handleTransitionEnd` :thinking: